### PR TITLE
Save and restore position of splitters

### PR DESCRIPTION
**Pull request by taeber**
This change remembers the position of both the "Towns/Castle" and "Heroes" list splitters in the Kingdom Overview dialog and restores them to their previous position when opened.

I tested the change using the [SAV file](https://github.com/ihhub/fheroes2/files/5195265/Full.elementals.zip) from #1609. I ensured both lists properly saved and restored. I also verified that the two players' previous splitter positions were independent of each other.

I used Visual Studio 2019 on Windows 10 x64 + SDL1 + Zlib (not that it should really matter in this case).

Fixes ihhub/fheroes2#1846

### CODE REVIEW

File: src/fheroes2/kingdom/kingdom_overview.cpp [hunk 1]
```diff
@@ -577,7 +577,11 @@ void Kingdom::OverviewDialog( void )
     RedrawFundsInfo( cur_pt, *this );
 
     StatsHeroesList listHeroes( dst_pt, heroes );
+    if ( listHeroes.GetSplitter().Max() > 0 )
```
Comment (ihhub) [inline src/fheroes2/kingdom/kingdom_overview.cpp:580]: Please cache `listHeroes.GetSplitter()` as a variable.

Comment (ihhub) [inline src/fheroes2/kingdom/kingdom_overview.cpp:580]: If max value is more than 0 doesn't guarantee that `prevHeroesSplitterPosition` is less than max value. The correct logic should be:
```
if ( heroesSplitter.Min() <= prevHeroesSplitterPosition && heroesSplitter.Max() >= prevHeroesSplitterPosition )
    heroesSplitter.MoveIndex( prevHeroesSplitterPosition );
```

File: src/fheroes2/kingdom/kingdom_overview.cpp [hunk 2]
```diff
@@ -577,7 +577,11 @@ void Kingdom::OverviewDialog( void )
...
+        listHeroes.GetSplitter().MoveIndex( prevHeroesSplitterPosition );
     StatsCastlesList listCastles( dst_pt, castles );
+    if ( listCastles.GetSplitter().Max() > 0 )
```
Comment (ihhub) [inline src/fheroes2/kingdom/kingdom_overview.cpp:583]: The same logic applies to castles as well.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @taeber , good job! Could you please take a look into given comments?

File: src/fheroes2/kingdom/kingdom_overview.cpp (hunk 1 above)
Comment (taeber) [reply src/fheroes2/kingdom/kingdom_overview.cpp:580]: Sorry that the intent of this code isn't clear. 

The `Max() > 0` check isn't to guarantee that `prevHeroesSplitterPosition` is less than max value (which isn't a precondition of `MoveIndex`). What it does is correctly handles the case when you lose your fifth hero by not moving the splitter away from the default, center position.

To see what I mean:
1. Start with 5 heroes.
2. Open the Kingdom Overview > Heroes dialog.
3. Scroll to the bottom.
4. Exit.
5. Somehow lose a hero.
6. Reopen Heroes dialog and notice the splitter remains in the center.

Without the check, the splitter suggest that there is more than 1 page and that scrolling is appropriate, which it's not for 4 heroes as they all fit on 1 page.

Does that make sense?

**Comment by taeber** [comment]
Sorry, I just realized that, while the scrollbar position does restore properly, the set of items is always the first four, so this changeset  does not fix the issue. In fact, it makes it worse as it can prevent you from scrolling down until you've scrolled up. I think the base `ListBox` class probably needs to have this save/restore functionality added to it, but I was trying to minimize the changes.

**Pull request closed without merging**